aspnet/migration/http-modules.rst

```diff
@@ -1,8 +1,584 @@
+
+.. literalinclude:: http-modules/sample/Asp.Net4/Asp.Net4/HttpHandlers/ReportHandler.cs
+	:language: c#
+	:linenos:
+	:emphasize-lines: 5, 7, 13-16
+	:lines: 1-19, 31-32
+
+In your ASP.NET 5 project, you would translate this to a middleware similar to this:
+
+.. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/ReportHandlerMiddleware.cs
+	:language: c#
+	:linenos:
+	:emphasize-lines: 7,9,13, 20-23, 29, 31, 33
+	:lines: 1-26, 38-47
+
+This middleware is very similar to the middleware corresponding to modules. The only real difference is that here there is no
+call to ``_next.Invoke(context)``. That makes sense, because the handler is at the end of the request pipeline, so there will be no next middleware to invoke.
+
+Migrating handler insertion into the request pipeline
+------------------------------------------------------
+In your ASP.NET 4 application, you will have inserted your HTTP handler in the request pipeline in your web.config file:
+
+.. literalinclude:: http-modules/sample/Asp.Net4/Asp.Net4/Web.config
+	:language: xml
+	:linenos:
+	:emphasize-lines: 7,8
+	:lines: 1-3, 29, 42-46, 48, 105
+
+You could convert this by adding your new handler middleware to the request pipeline in your ``Startup`` class, similar to 
+middleware converted from modules.
+
+Problem is that this would send all requests to your new handler middleware. However, you only want requests with a given
+extension to reach your middleware. That would give you the same situation you had with your HTTP handler.
+
+A solution is to branch the pipeline for requests with a given extension, using the ``MapWhen`` extension method.
+This happens in the same ``Configure`` method where you add the other middleware:
+
+.. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Startup.cs
+	:language: c#
+	:linenos:
+	:lines: 1-2, 12-15, 46-48, 54-55, 82-87, 105, 109, 110
+	:emphasize-lines: 12-17
+
+``MapWhen`` takes these parameters:
+
+1. A lambda that takes the ``HttpContext`` and returns ``true`` if the request should go down the branch.
+   This means you can branch requests not just based on their extension, but also on request headers, query string parameters, etc.;
+
+2. A lambda that takes an ``IApplicationBuilder`` and adds all the middleware for the branch.
+   This means you can add additional middleware to the branch in front of your handler middleware.
+
+The middleware that was added to the pipeline before the branch will be invoked for both requests going down the branch
+and for requests not going down the branch.
+
+Loading middleware options using the options pattern
+----------------------------------------------------
+
+Some modules and handlers have configuration options. In your ASP.NET 4 application, you would have stored these options in your web.config file.
+However, in ASP.NET 5 you can no longer do that.
+
+The new ASP.NET 5 :ref:`configuration system <fundamentals-configuration>` gives you these options to solve this:
```

Comment:
Couldn't get this to work. Tried

:doc:`../fundamentals/configuration#fundamentals-configuration`
:doc:`http://docs.asp.net/en/latest/fundamentals/configuration.html#fundamentals-configuration`
:doc:`<http://docs.asp.net/en/latest/fundamentals/configuration.html#fundamentals-configuration>`
:doc:`<../fundamentals/configuration.html#fundamentals-configuration>`

none of which produced a link. How do you link to a section within another document with :doc: ?